src/rewriters/CustomRewriter.ts

```diff
@@ -0,0 +1,73 @@
+import { ASTNode } from 'graphql';
+import { NodeAndVarDefs } from '../ast';
+import Rewriter, { RewriterOpts, Variables } from './Rewriter';
+
+interface CustomRewriterOpts extends RewriterOpts {
+  matchesFn?: (nodeAndVarDefs: NodeAndVarDefs, parents: ReadonlyArray<ASTNode>) => boolean;
+  rewriteQueryFn?: (nodeAndVarDefs: NodeAndVarDefs, variables: Variables) => NodeAndVarDefs;
+  rewriteVariablesFn?: (nodeAndVarDefs: NodeAndVarDefs, variables: Variables) => Variables;
+  rewriteResponseFn?: (
+    response: any,
+    key: string,
+    index?: number,
+    nodeMatchAndParents?: ASTNode[]
+  ) => NodeAndVarDefs;
+}
+
+/**
+ * A Custom rewriter with its Rewriter functions received as arguments.
+ * This Rewriter allows users to write their own rewriter functions.
+ */
+class CustomRewriter extends Rewriter {
```

Comment:
What benefit does this have over just extending the `Rewriter` class directly to make a custom rewriter?

Reply:
That was my initial thought, however it's simpler not having to created a class, specially when one isn't working in typescript.

Reply:
Yeah that's fair, it's probably easier to document too